homeassistant/components/vulcan/calendar.py

```diff
@@ -55,20 +59,17 @@ async def async_setup_entry(
     )
 
 
-class VulcanCalendarEventDevice(CalendarEventDevice):
-    """A calendar event device."""
+class VulcanCalendarEntity(CalendarEntity):
+    """A calendar entity."""
 
-    def __init__(self, client, data, entity_id):
-        """Create the Calendar event device."""
+    def __init__(self, client, data, entity_id) -> None:
+        """Create the Calendar entity."""
         self.student_info = data["student_info"]
-        self.data = VulcanCalendarData(
-            client,
-            self.student_info,
-            self.hass,
-        )
-        self._event = None
+        self._event: CalendarEvent | None = None
+        self.client = client
         self.entity_id = entity_id
         self._unique_id = f"vulcan_calendar_{self.student_info['id']}"
+        self.service_available = True
```

Comment:
Shouldn't we implement the `available` property or attribute?

https://developers.home-assistant.io/docs/core/entity